lib/session.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize,Lint/MissingCopEnableDirective
+
+module AutoHCK
+  class Session
+    extend T::Sig
+    include Helper
+
+    def self.save(workspace_path, options)
+      File.write("#{workspace_path}/session.json", compose_session_json(options, workspace_path))
+    end
+
+    def self.load(cli)
+      session = Models::Session.from_json_file("#{session_path(cli)}/session.json")
+
+      backup_cli = cli.clone
```

Comment:
This `clone` can be removed by modifying `session` according to `cli` and then overwriting `cli`.